### PR TITLE
Make it so that the drawer text wont show when the drawer is closed

### LINKED ISSUE #126: drawer multiple nav groups example with string not hidden when collapsed
#### Describe the bug
We have 2 title nav items without icons and when drawer is collapsed the first character is displayed. 

#### What is the expected behavior?
To hide the 2 single title nav items on collapse or pass in icon for title to display icon when drawer is collapsed.

#### What are the steps to reproduce?
1. Go to https://pxblue-components.github.io/angular-dev/?path=/story/components-drawer--with-multiple-nav-groups
2. Click on drawer menu to collapse.

#### Screenshots or links to minimum reproduction example
image

### PR DESCRIPTION
**Pull request by emclaug2**
Fixes #126 

#### Changes proposed in this Pull Request:
- When the drawer is closed, the nav-items with no icons are set to hide.

An alternative solution would be to make this drawer non-closable. 

#### Screenshots:
image

image

### CODE REVIEW

**Comment by joebochill** [comment]
@huayunh is this the desired behavior? Or do we want to do something like ellipse the text?

**Comment by emclaug2** [comment]
> @huayunh is this the desired behavior? Or do we want to do something like ellipse the text?

I would imagine that we shouldn't be encouraging users to have text-only nav items with a persistent drawer variant.

**Comment by emclaug2** [comment]
Another potential solution here is to just add icons to the second nav group.

**Comment by joebochill** [comment]
Yes, we should encourage them to use icons at the top level with persistent drawer, but we also should define what the behavior will be (clipping, hiding, ellipsing, etc.) if they don't.

**Comment by huayunh** [comment]
(1) Whoever tries to design a product using a persistent drawer without icons has a terrible design taste
(1.1) using a mix of menu items with & without icons on permanent & temporary drawers might be a bad idea too, but not as horrible as (1)
(2) storybook should probably show our best design practice, so set this drawer to permanent or temporary would be a good idea for this story in particular
(3) if they somehow desperately want to use a persistent drawer but with iconless items, what I see in the screenshots above is acceptable; if you want to go fancier, you can shrink the text to caption but font-weight 600 (can be hard to see under 400), and use an ellipsis if still overflowing

image

**Comment by joebochill** [comment]
I'm fine with just hiding it...simplest and easiest to do. Primary design recommendation will be to avoid doing this when possible. Since we haven't introduced the variants yet by this example (and persistent is the default), I'd be fine with just adding icons to the second group in this example. I'm also not opposed to making the variant permanent, it just might be a bit weird for users if they haven't read the readme.

File: demos/storybook/stories/drawer/with-multiple-nav-groups.stories.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export const withMultiNavGroups = (): any => ({
               <pxb-spacer *ngIf="spacer"></pxb-spacer> 
               <pxb-drawer-nav-group [title]="groupTitle2" [divider]="true">
                  <pxb-drawer-nav-item *ngFor="let navItem of navItems2"
-                    [title]="navItem.title"
+                    [title]="state.open ? navItem.title : ''"
```
Comment (joebochill) [inline demos/storybook/stories/drawer/with-multiple-nav-groups.stories.ts:38]: Wait a second...shouldn't we be making this change in the component behavior?

**Comment by joebochill** [comment]
Assuming we'll also want to make this change for React

File: demos/storybook/stories/drawer/with-multiple-nav-groups.stories.ts (hunk 1 above)
Comment (emclaug2) [reply demos/storybook/stories/drawer/with-multiple-nav-groups.stories.ts:38]: This story needs to updated to add icons to the second nav group.

Comment (joebochill) [reply demos/storybook/stories/drawer/with-multiple-nav-groups.stories.ts:38]: Yes. I think we also need to update the component too. The bug in question is 'labels getting chopped off when collapsed' - so shouldn't we update the component to hide the labels on collapse if there are no icons?

**Comment by emclaug2** [comment]
I'm still unsure what our recommended approach is supposed to be, should we hide the text or shrink the text?

**Comment by joebochill** [comment]
> I'm still unsure what our recommended approach is supposed to be, should we hide the text or shrink the text?

Definitely hide the text

**Comment by emclaug2** [comment]
Drawer no icon

**Comment by huayunh** [comment]
Maybe opacity animation transition instead?

**Comment by emclaug2** [comment]
> Maybe opacity animation transition instead?

And keep the ellipsis?

**Comment by emclaug2** [comment]
> > Maybe opacity animation transition instead?
> 
> And keep the ellipsis?

The ellipsis is something that affects all nav items at the moment; this would be a change for all items.

**Comment by joebochill** [comment]
If it's already doing the ellipsis, I would leave it alone

**Comment by emclaug2** [comment]
> If it's already doing the ellipsis, I would leave it alone

I tried to slow down the Drawer animation, it's using ellipsis currently.  
Ellipis

**Comment by emclaug2** [comment]
d 55

**Comment by emclaug2** [comment]
React has the same bug, made a new issue for it.

**Pull request merged**